### ISSUE TITLE
alt-svc build error on ubuntu 18.04: missing struct tm

### ISSUE THREAD

**Issue by bagder** [curl/curl#9989]
The vexxhost CI [build](https://object-storage.public.mtl1.vexxhost.net/swift/v1/bfd521072e894ebb99e66f72619daa8a/zuul-logs-curl_d8b/9987/98bdab82e8a83c0dd27f126aab398902c956e745/check/curl-novalgrind-boringssl-with-openssl/d8b184f/job-output.txt) with BoringSSL shows this build error of altsvc:

~~~
Making all in lib
make[1]: Entering directory '/home/zuul/src/github.com/curl/curl/lib'
make  all-am
make[2]: Entering directory '/home/zuul/src/github.com/curl/curl/lib'
  CC       libcurl_la-altsvc.lo
In file included from altsvc.c:36:
parsedate.h:30:44: error: â€˜struct tmâ€™ declared inside parameter list will not be visible outside of this definition or declaration [-Werror]
 CURLcode Curl_gmtime(time_t intime, struct tm *store);
                                            ^~
altsvc.c: In function â€˜altsvc_outâ€™:
altsvc.c:233:13: error: storage size of â€˜stampâ€™ isnâ€™t known
   struct tm stamp;
             ^~~~~
altsvc.c: In function â€˜Curl_altsvc_parseâ€™:
altsvc.c:592:36: error: implicit declaration of function â€˜timeâ€™; did you mean â€˜niceâ€™? [-Wimplicit-function-declaration]
             as->expires = maxage + time(NULL);
                                    ^~~~
                                    nice
cc1: all warnings being treated as errors
Makefile:1758: recipe for target 'libcurl_la-altsvc.lo' failed
make[2]: *** [libcurl_la-altsvc.lo] Error 1

**Comment by bagder** [comment]
We build lots of builds on this platform but this is mysteriously enough the only one that gets this error.

**Comment by bagder** [comment]
Turns out there's a `time.h` file in the openssl include dir that we managed to get to "shadow" the real `time.h` header due to how we setup the include path.

**Comment by cfc4n** [comment]
the same issue on ubuntu 20.04
```shell
git clone https://github.com/curl/curl.git
autoreconf -fi
LIBS=-lpthread ./configure --with-ssl=/home/cfc4n/project/boringssl
```

output:
```shell
make
Making all in lib
make[1]: Entering directory '/home/cfc4n/project/curl/lib'
make  all-am
make[2]: Entering directory '/home/cfc4n/project/curl/lib'
  CC       libcurl_la-altsvc.lo
In file included from altsvc.c:36:
parsedate.h:30:44: warning: ‘struct tm’ declared inside parameter list will not be visible outside of this definition or declaration
   30 | CURLcode Curl_gmtime(time_t intime, struct tm *store);
      |                                            ^~
altsvc.c: In function ‘altsvc_out’:
altsvc.c:233:13: error: storage size of ‘stamp’ isn’t known
  233 |   struct tm stamp;
      |             ^~~~~
altsvc.c: In function ‘Curl_altsvc_parse’:
altsvc.c:592:36: error: implicit declaration of function ‘time’ [-Werror=implicit-function-declaration]
  592 |             as->expires = maxage + time(NULL);
      |                                    ^~~~
cc1: some warnings being treated as errors
make[2]: *** [Makefile:2067: libcurl_la-altsvc.lo] Error 1
make[2]: Leaving directory '/home/cfc4n/project/curl/lib'
make[1]: *** [Makefile:1433: all] Error 2
make[1]: Leaving directory '/home/cfc4n/project/curl/lib'
make: *** [Makefile:1239: all-recursive] Error 1
```

**Comment by cfc4n** [comment]
why close PR #9988 ?  It resolved  my issue.

**Comment by jzakrzewski** [comment]
> why close PR #9988 ? It resolved my issue.

This is the way this project merges PRs - the fix is commited in 42aa4e1a70b13b0574e6b685fc0f60cea3fc114f as the closing message statets.

**Comment by cfc4n** [comment]
got it, thanks.
